chichiwang/exercise-js
# Add Eslint

## Description
Linked to Issue: #3
Add eslint to the project to enforce code style and catch careless errors. Ensure that it works with Jest and starts with a common base configuration (`eslint-config-airbnb`). Add a script `lint` to `package.json` to run the linter against the codebase.

## Changes
* Add the following `devDependencies` to `package.json`:
  * `eslint`
  * `eslint-config-airbnb`
  * `eslint-plugin-jest`
* Add script `lint` to `package.json` to run eslint against the entire codebase
* Update `babel.config.js` to conform to lint rules
* Add exception for file `server/__tests__/index.test.js` from the rule `global-require`
  *  A non-global require is necessary to reload the file under test (`server/index.mjs`) multiple times to execute tests

## Steps to QA
* Pull down and switch to this branch `chichi/add-eslint`
* Run `npm run lint` to ensure passing linter run
* Update a `.js`/`.mjs` file to contain a lint violation (ex: remove a trailing comma on a multi-line array/object)
  * Run `npm run lint` to ensure a failing linter run, with the newly added violation listed
